source/renderer/app/components/staking/stake-pools/hooks/useCreateColumns.tsx

```diff
@@ -0,0 +1,261 @@
+import React, { useMemo } from 'react';
+import classNames from 'classnames';
+import BigNumber from 'bignumber.js';
+import moment from 'moment';
+import { FormattedHTMLMessage } from 'react-intl';
+import { Column } from 'react-table';
+import { PopOver } from 'react-polymorph/lib/components/PopOver';
+import { PoolPopOver } from '../../widgets/PoolPopOver';
+import { Intl } from '../../../../types/i18nTypes';
+import styles from '../StakePoolsTable.scss';
+import StakePool from '../../../../domains/StakePool';
+import {
+  getColorFromRange,
+  getSaturationColor,
+} from '../../../../utils/colors';
+import {
+  formattedWalletAmount,
+  toFixedUserFormat,
+} from '../../../../utils/formatters';
+import { messages } from '../StakePoolsTable.messages';
+
+type UseCreateColumnsArgs = {
+  currentTheme: string;
+  showWithSelectButton?: boolean;
+  onSelect?: (poolId: string) => void;
+  containerClassName: string;
+  numberOfRankedStakePools: number;
+  onOpenExternalLink: (url: string) => void;
+  intl: Intl;
+};
+
+export const useCreateColumns = ({
+  numberOfRankedStakePools,
+  intl,
+  currentTheme,
+  onOpenExternalLink,
+  onSelect,
+  containerClassName,
+  showWithSelectButton,
+}: UseCreateColumnsArgs) =>
+  useMemo<Column<StakePool>[]>(
```

Comment:
Right now properties `id` and `accessor` can be assigned with any string not necessary corresponding to the `StakePool` properties. I initially thought that you will create an enum for all fields and use it for `id` and `accessor` properties bellow. However having the `StakePool` class we can use it to restrict possible values of those properties by adjusting slightly the type passed to `useMemo`:
```suggestion
type StakePoolColumn = Column<StakePool> & {
  id: keyof StakePool,
  accessor: keyof StakePool,
};
// ...
  useMemo<StakePoolColumn[]>(
```